RearrangingCarsTest.java

```diff
@@ -90,6 +92,32 @@ public void testDifferentLengthInputs() {
         Assert.assertNull(moves);
     }
 
+    @org.junit.Test
+    public void testOptimisedRearrange() {
+        for (int size = 5; size < 50; size++) {
+            int[] originalOrder = IntStream.rangeClosed(0, size - 1).toArray();
+            int[] finalOrder = originalOrder.clone();
+            shuffleArray(finalOrder);
```

Comment:
Be careful! If I am not mistaken, this is not actually doing anything (in terms of things you need). Basically, shuffleArray should return an array, and you should assign finalOrder to shuffleArray(finalOrder). So unless finalOrder was a pointer, there is no way this line shuffles the array.